supportedBackends/aws/src/main/resources/sidecar/sidecar

```diff
@@ -0,0 +1,117 @@
+          ;;
+        *)
+          aws s3 cp "${path}${file}" "$AWS_CROMWELL_CALL_ROOT/${localDiskRelativePath}${file}"
+      esac
+    elif [ "$action" = "MOVED_FROM" ]; then
+      case $path in
+        ${AWS_CROMWELL_LOCAL_DISK}/${exclude}*)
+          echo skipping change in excluded directory
+          ;;
+        *)
+          # The file was removed from the local disk, so we'll remove it
+          # from s3 as well.
+          aws s3 rm "$AWS_CROMWELL_CALL_ROOT/${localDiskRelativePath}${file}"
+      esac
+    fi
+  done
+}
+
+###########################################################################
+# The general plan is to do a sync of all workflow files into the container
+# We share the same volumes/mount points of the original, so we can go to
+# town here. Generally, we assume all *inputs* come from
+# AWS_CROMWELL_WORKFLOW_ROOT
+# All *outputs* will eventually end up in AWS_CROMWELL_CALL_ROOT.
+# However, within the container, this root is effectively AWS_CROMWELL_LOCAL_DISK
+#
+# The trick here is that the inputs are also landed inside
+# AWS_CROMWELL_LOCAL_DISK. So we need to exclude the inputs from being
+# synced (both real time and in batch at the end), so we do some fancy
+# stuff for that
+###########################################################################
+# call root example: s3://cromwell-aws/cromwell-execution/...
+# localRoot example: /cromwell_root/cromwell-aws/cromwell-execution...
+# AWS_CROMWELL_PATH example: ...
+localCallRoot="$AWS_CROMWELL_LOCAL_DISK"/"${AWS_CROMWELL_CALL_ROOT#"s3://"}"
+localWorkflowRoot="$AWS_CROMWELL_LOCAL_DISK"/"${AWS_CROMWELL_WORKFLOW_ROOT#"s3://"}"
+suffix="/*"
+exclude=${AWS_CROMWELL_CALL_ROOT#"s3://"}
+exclude=${exclude%%$suffix}
+echo "syncing files from s3"
+echo "source ${AWS_CROMWELL_WORKFLOW_ROOT}, dest $localWorkflowRoot"
+aws s3 sync "$AWS_CROMWELL_WORKFLOW_ROOT" "$localWorkflowRoot"
+echo "Resuming container"
+docker unpause "$AWS_CROMWELL_TASK_DOCKER_ID"
+# Place watch for the rc file
+waitfor "$AWS_CROMWELL_LOCAL_DISK" "${AWS_CROMWELL_RC_FILE#"${AWS_CROMWELL_LOCAL_DISK}/"}"
+
+# We'll pause the task container while we do our work. This will prevent
+# the container from exiting before files are out on S3. Also, the ecs agent
+# has some cleanup, and we don't want a race condition with that process
+docker pause "$AWS_CROMWELL_TASK_DOCKER_ID"
+
+###########################################################################
+# Now we will do the final sync. If we've done the job right, this sync
+# should only copy the rc file in place to s3, since all the other files
+# should have been caught during the task run with our inotifywatch magic(tm)
+###########################################################################
+echo "syncing files with s3"
+echo aws s3 sync "$AWS_CROMWELL_LOCAL_DISK" "$AWS_CROMWELL_CALL_ROOT" --exclude "${exclude}/*"
+aws s3 sync "$AWS_CROMWELL_LOCAL_DISK" "$AWS_CROMWELL_CALL_ROOT" --exclude "${exclude}/*"
```

Comment:
Could you use this exclude variable to also exclude everything that has not been declared a job output by the author ? This would prevent from delocalizing a potentially large number of unwanted temporary files to S3.

Reply:
As the design sits today I don't have that information in the sidecar, so I can't exclude things not explicitly declared. I'm going to take a look today and see if I can get more specific with both inputs and outputs.